.github/workflows/equinix_metal_flow.yml

```diff
@@ -0,0 +1,149 @@
+          sudo cat > inventory.yml << EOF
+          all:
+            children:
+              servers:
+                hosts:
+                  localhost:
+                vars:
+                  firewall:
+                    - service: pmcd
+                      state: enabled
+                  metrics_retention_days: 7
+              metrics_monitor:
+                hosts:
+                  localhost:
+                vars:
+                  firewall:
+                    - service: grafana
+                      state: enabled
+                  metrics_graph_service: yes
+                  metrics_query_service: yes
+                  metrics_retention_days: 7
+                  metrics_monitored_hosts: "{{ groups['servers'] }}"
+          EOF
+          cat > metrics.yml << EOF
+          - name: Use metrics system role to configure PCP metrics recording
+            hosts: servers
+            roles:
+              - redhat.rhel_system_roles.metrics
+              - redhat.rhel_system_roles.firewall
+            become: true
+
+          - name: Use metrics system role to configure Grafana
+            hosts: metrics_monitor
+            roles:
+              - redhat.rhel_system_roles.metrics
+              - redhat.rhel_system_roles.firewall
+            become: true
+          EOF
+          ansible-playbook -i inventory.yml metrics.yml
+
+          set -o pipefail
+          set -e
+          # install podman
+          sudo dnf -y install podman
+          # wget kepler systemd unit file and save it to /etc/systemd/system/kepler.service
+          sudo wget https://raw.githubusercontent.com/sustainable-computing-io/kepler/main/packaging/rpm/container-kepler.service -O /etc/systemd/system/container-kepler.service
+          # start kepler service
+          sudo systemctl daemon-reload
+          sudo systemctl enable --now container-kepler
+          # curl kepler port 8888 with 10 seconds timeout and retry 5 times; if it fails, exit with error
+          for i in {1..5}; do 
+            echo "Attempt $i"
+            # curl and get http code
+            ret=$(curl -s -o /dev/null -w "%{http_code}" http://localhost:8888/metrics --max-time 10 --connect-timeout 10 || true)
+            if [ ${ret} -eq 200 ]; then 
+              break 
+            fi
+            sleep 5
+          done
+          if [ ${ret} -ne 200 ]; then 
+            echo "Kepler did not start in time"
+            # exit 1
+          fi
+          # dump kepler metrics
+          echo "Kepler started successfully, now dumping metrics"
+          curl -s http://localhost:8888/metrics |grep ^kepler_
+          # install openmetrics
+          echo "Installing openmetrics"
+          sudo yum install pcp-pmda-openmetrics -y
+          cd /var/lib/pcp/pmdas/openmetrics/; sudo echo "http://localhost:8888/metrics" > config.d/kepler.url; sudo ./Install
```

Comment:
should we considering using golang or other test code? I suppose we have part of them at e2e folders, maybe we can enhance them by env variable?